gnomad_constraint/utils/constraint_basics.py

```diff
@@ -165,20 +167,221 @@ def get_proportion_observed_by_coverage(
+
+    high_coverage_ht = annotate_variant_types(high_coverage_ht, not trimers)
+    plateau_models = build_plateau_models(high_coverage_ht, weighted=weighted)
+
+    high_coverage_scale_factor = all_high_coverage_ht.aggregate(
+        hl.agg.sum(all_high_coverage_ht.variant_count)
+        / hl.agg.sum(
+            all_high_coverage_ht.possible_variants * all_high_coverage_ht.mu_snp
         )
     )
-    return (
-        t.annotate_rows(vep=annotation)
-        if isinstance(t, hl.MatrixTable)
-        else t.annotate(vep=annotation)
+
+    all_low_coverage_ht = coverage_ht.filter(
+        (coverage_ht.exome_coverage < cov_cutoff) & (coverage_ht.exome_coverage > 0)
+    )
+
+    low_coverage_ht = all_low_coverage_ht.group_by(
+        log_coverage=hl.log10(all_low_coverage_ht.exome_coverage)
+    ).aggregate(
+        low_coverage_obs_exp=hl.agg.sum(all_low_coverage_ht.variant_count)
+        / (
+            high_coverage_scale_factor
+            * hl.agg.sum(
+                all_low_coverage_ht.possible_variants * all_low_coverage_ht.mu_snp
+            )
+        )
+    )
+    coverage_model = build_coverage_model(low_coverage_ht)
+    # TODO: consider weighting here as well
+
+    return coverage_model, plateau_models
+
+
+def build_plateau_models(
+    ht: hl.Table,
+    weighted: bool = False,
+    pop_anal: bool = True,
+    pops: Tuple[str] = POPS,
+) -> Dict[str, Tuple[float, float]]:
+    """
+    Calibrate high coverage model (returns intercept and slope).
+
+    The function fits two models, one for CpG transitions and one for the remainder of sites, to calibrate
+    from the mutation rate to proportion observed in total and in each population.
+
+    .. note::
+        The following annotations should be present in `ht`:
+            - observed_variants - observed variant counts for each combination of 'context', 'ref', 'alt', 'methylation_level', 'mu_snp
+            - observed_variants_{pop} (where pop is each population) - observed variant counts for each population
+            - possible_variants - possible variant counts for each combination of 'context', 'ref', 'alt', 'methylation_level', 'mu_snp
+            - cpg - whether it's a cpg site or not
+            - mu_snp - mutation rate
+
+    :param ht: High coverage Table.
+    :param pop_anal: Whether to build models for each population. Defaults to True.
+    :param pops: List of populations. Defaults to `POPS`.
+    :param weighted: Whether to generalize the model to weighted least squares using 'possible_variants'.
+    :return: A Dictionary of intercepts and slopes for the full plateau models.
+    """
+    pop_lengths = get_all_pop_lengths(ht, pops)
+
+    plateau_models = build_plateau_models_total(ht, weighted=weighted)
+
+    if pop_anal:
+        arg_expr = {}
+        for length, pop in pop_lengths:
+            plateau_model = build_plateau_models_pop(ht, pop, length, weighted=weighted)
```

Comment:
the variable names (`pleatu_models`, `plateau_models`) are super similar; it might be helpful to make this one `pop_plateau_model` or something more descriptive